### PR TITLE
[OSDEV-360] Facility Search. Make it easier to search for groups of related sectors (BE)

### PR DESCRIPTION
**Pull request by mazursasha1990**
[OSDEV-360](https://opensupplyhub.atlassian.net/browse/OSDEV-360) - Facility Search. Make it easier to search for groups of related sectors (BE)

 The following changes have been implemented:

-  A new table, `api_sectorgroup`, has been introduced and populated with sector group names. 
-  A new field named `groups` has been added to the `Sector` model to establish a many-to-many relationship between the `Sector` model and the `SectorGroup` model.
- `SectorGroup` was added to the admin page.

[OSDEV-360]: https://opensupplyhub.atlassian.net/browse/OSDEV-360?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/django/api/migrations/0152_add_sector_group_table.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        "Education, Research, Services",
+        "Other",
+    ]
+
+    for name in groups_names:
+        SectorGroup.objects.create(name=name)
+
+
+def revert_populate_sector_groups(apps, schema_editor):
+    SectorGroup = apps.get_model('api', 'SectorGroup')
```
Comment (roman-stolar) [inline src/django/api/migrations/0152_add_sector_group_table.py:39]: I'm not sure that this revert is required as on revert action it should just delete model/table **SectorGroup**.

File: src/django/api/migrations/0153_associate_sectors_with_groups.py [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    sectors = Sector.objects.all()
+    for sector in sectors:
+        for group_name, sector_names in sector_groups.items():
+            if sector.name in sector_names:
+                group_instance = SectorGroup.objects.get(name=group_name)
+                sector.groups.add(group_instance)
+
+
+def revert_associate_sectors_with_groups(apps, schema_editor):
+    for sector in Sector.objects.all():
```
Comment (roman-stolar) [inline src/django/api/migrations/0153_associate_sectors_with_groups.py:214]: Similar as above, if we remove the field from model/table the data will be cleared automatically

File: src/django/api/migrations/0152_add_sector_group_table.py (hunk 1 above)
Comment (mazursasha1990) [reply src/django/api/migrations/0152_add_sector_group_table.py]: Replaced it to `migrations.RunPython.noop`.

File: src/django/api/migrations/0153_associate_sectors_with_groups.py (hunk 2 above)
Comment (mazursasha1990) [reply src/django/api/migrations/0153_associate_sectors_with_groups.py]: Replaced it to `migrations.RunPython.noop`.

**Review by roman-stolar — APPROVED** [review]
LGTM

File: src/django/api/models/__init__.py [hunk 3]
```diff
@@ -68,6 +68,7 @@
 from .production_type import ProductionType
 from .request_log import RequestLog
 from .sector import Sector
+from .sector_group import SectorGroup
 from .source import Source
```
Comment (killev) [inline src/django/api/models/__init__.py:72]: Let's not add Models to __init__.py to avoid disambiguates

Comment (mazursasha1990) [reply src/django/api/models/__init__.py]: Added import of SectorGroup to the admin.py

**Review by killev — APPROVED** [review]
LGTM, Approved!

**Review by roman-stolar — APPROVED** [review]
LGTM

**Review by roman-stolar — APPROVED** [review]

**Review by killev — APPROVED** [review]
LGTM. Approved

**Pull request merged**